src/etc/inc/interfaces.inc

```diff
@@ -2357,6 +2358,87 @@ function interface_virtual_create($interface)
     }
 }
 
+function interface_dslite_configure($verbose, $parent = 'wan', $aftr)
+{
+    global $config;
+
+    // Should be tested outside, but let's make sure we won't mess around
+    if (!interface_dslite_is_enabled($parent)) {
+        return;
+    }
+
+    $wanconfig = $config['interfaces'][$parent];
+    $mtu = !empty($wanconfig['dslite_tunnel_mtu']) ? $wanconfig['dslite_tunnel_mtu'] : '';
+    $ip = get_interface_ipv6($parent);
+
+    if ($verbose) {
+        log_error(sprintf("Starting DS-Lite tunnel with AFTR '%s' on interface %s with external IP '%s'", $aftr, $parent, $ip));
+    }
+
+    $gifif = legacy_interface_create('gif', "{$parent}_dslite");
+
+    $pmtu = '';
+    if (!empty($mtu)) {
+        $pmtu = sprintf('mtu %s', $mtu);
+    }
+
+    mwexecf('/sbin/ifconfig %s inet6 tunnel %s %s %s -accept_rtadv ifdisabled', array($gifif, $ip, $aftr, $pmtu));
+    mwexecf("/sbin/ifconfig %s inet 192.0.0.2 192.0.0.1 netmask 255.255.255.248", array($gifif));
+
+    $dslite_tunnel_file = sprintf("/tmp/dslite_%s.state", $parent);
+    @file_put_contents($dslite_tunnel_file, $gifif);
+
+    $dslite_aftr_file = sprintf("/tmp/dslite_%s_aftr.state", $parent);
+    @file_put_contents($dslite_aftr_file, $aftr);
```

Comment:
It should be possible to read it back from ifconfig as long as there is a connection. If the tunnel drops for any reason I'm not sure the AFTR address would still be readable. I think a safe fallback would be to try to read it from ifconfig and if not available recreate the tunnel using the AFTR address in the file. Not everything I want to implement is done yet, for example tunnel monitoring and possibly recreating it :)

Reply:
Is there any way we can check if the address is and stays in ifconfig? If it does, I can add the option to legacy_interfaces_details() so we can always stick to the interface and prevent files from getting async.

Reply:
Not sure, my AFTR connection seems to be pretty stable. Normally when a reconnection (in the Fritzbox) happens you get a new external IP address (which is still shared by multiple people but is "new" to you). I didn't have this situation yet, at least not that I realized. I think setting up the gateway monitoring (for 192.0.0.1) might give some hint if we actually need to recreate the interface at any time. Whenever I apply changes to the/a wan interface, I should get a new DHCPv6 request, shouldn't I?

Reply:
Sure thing nothing easier than that, however I'd imagine it is already available, since the 4in6 is most probably just a gif tunnel :-)

Anyhow here it is:
```
gif0: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> metric 0 mtu 1460
	options=80000<LINKSTATE>
	capabilities=80000<LINKSTATE>
	tunnel inet6 2a02:908:0000:0000:0000:0000:b863 --> 2a02:908::13:4000
	inet 192.0.0.2 --> 192.0.0.1  netmask 0xfffffff8
	nd6 options=29<PERFORMNUD,IFDISABLED,AUTO_LINKLOCAL>
	groups: gif
```

Just changed a few bits in my IPv6 ;-) and the 2a02:908::13:4000 is the AFTR address in my case. I guess you might not be able to reach / ping that address though. As far as I know, it's only reachable from the inside of the Unitymedia network.

Reply:
ok, then tunnel wasn't parsed yet, with  https://github.com/opnsense/core/commit/0ac2af1c1a83bc975c5c8f6ff86d5cd49c0a0291 you can easily extract it from `legacy_interfaces_details()`

>            [tunnel] => Array
>                (
>                    [proto] => inet6
>                    [src_addr] => 2a02:908:0000:0000:0000:0000:b863
>                    [dest_addr] => 2a02:908::13:4000
>                )